Add some clarifications about control pre-provisioned CP nodes (#4133)
mesosphere/dcos-docs-site: pages/dkp/konvoy/2.2/choose-infrastructure/pre-provisioned/define-control-plane-endpoint/index.md

enterprise: false
---

> A control plane should have three, five, or seven nodes, so that it can remain available if one, two, or three nodes fail. A control plane with one node should not be used in production.

Reviewer: ```suggestion
A control plane should have three, five, or seven nodes, so it can remain available if one, two, or three nodes fail. A control plane with one node should not be used in production.
```

## External load balancer

It is recommended that an external load balancer be the control plane endpoint. To distribute request load among the control plane machines, configure the load balancer to send requests to all the control plane machines. Configure the load balancer to send requests only to control plane machines that are responding to API requests.

## Built-in virtual IP

If an external load balancer is not available, use the built-in virtual IP. The virtual IP is _not_ a load balancer; it does not distribute request load among the control plane machines. However, if the machine receiving requests does not respond to them, the virtual IP automatically moves to another machine.

## Single-Node control plane

> A control plane with one node can use its single node as the endpoint. Which means that neither an external load balancer, nor the built-in virtual IP is required. But, a single-node control plane configuration should not be used in production becuase it is not possible to upgrade the control plane if you only have one node. This is because CAPI does not perform in-situ upgrades, it instead replaces machines. If you want to upgrade your cluster, use the default <code>--control-plane-replicas</code> setting of 3.

Reviewer: ```suggestion
A control plane with one node can use its single node as the endpoint, so you will not require an external load balancer, or a the built-in virtual IP. However, do not use a single-node control plane configuration in production, as it is not possible to upgrade the control plane if you only have one node. This is due to the fact that CAPI does not perform in-situ upgrades, instead, it replaces machines. If you want to upgrade your cluster, use the default <code>--control-plane-replicas</code> setting of 3.
```
